jpa-criteria/impl/src/main/java/com/blazebit/persistence/criteria/impl/BlazeCriteriaBuilderImpl.java

```diff
@@ -1376,4 +1377,9 @@ public BlazeOrderedSetAggregateFunctionExpression<String> listagg(Expression<Str
     public BlazeOrderedSetAggregateFunctionExpression<String> listaggDistinct(Expression<String> expression, Expression<String> separator) {
         return new OrderedSetAggregationFunction<>(this, String.class, "LISTAGG", true, expression, separator);
     }
+
+    @Override
+    public <T> BlazeCriteriaSetQuery<T> union(BlazeCriteriaQuery<T> a, BlazeCriteriaQuery<T> b) {
+        return new BlazeCriteriaSetBuilderImpl<>(this, a.getResultType(), (BlazeCriteriaQueryImpl<T>) a,  (BlazeCriteriaQueryImpl<T>) b);
```

Comment:
Not sure how to go from API to implementation here, to get access to the internals.

Reply:
Casting stuff is totally fine here IMO.